paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkCatalog.java

```diff
@@ -315,48 +320,77 @@ public boolean dropTable(Identifier ident) {
         }
     }
 
-    private SchemaChange toSchemaChange(TableChange change) {
+    private List<SchemaChange> toSchemaChange(TableChange change) {
+        final List<SchemaChange> schemaChanges = new ArrayList<>();
         if (change instanceof TableChange.SetProperty) {
             TableChange.SetProperty set = (TableChange.SetProperty) change;
             validateAlterProperty(set.property());
-            return SchemaChange.setOption(set.property(), set.value());
+            schemaChanges.add(SchemaChange.setOption(set.property(), set.value()));
+            return schemaChanges;
         } else if (change instanceof TableChange.RemoveProperty) {
             TableChange.RemoveProperty remove = (TableChange.RemoveProperty) change;
             validateAlterProperty(remove.property());
-            return SchemaChange.removeOption(remove.property());
+            schemaChanges.add(SchemaChange.removeOption(remove.property()));
+            return schemaChanges;
         } else if (change instanceof TableChange.AddColumn) {
             TableChange.AddColumn add = (TableChange.AddColumn) change;
             validateAlterNestedField(add.fieldNames());
             SchemaChange.Move move = getMove(add.position(), add.fieldNames());
-            return SchemaChange.addColumn(
-                    add.fieldNames()[0],
-                    toPaimonType(add.dataType()).copy(add.isNullable()),
-                    add.comment(),
-                    move);
+            schemaChanges.add(
+                    SchemaChange.addColumn(
+                            add.fieldNames()[0],
+                            toPaimonType(add.dataType()).copy(add.isNullable()),
+                            add.comment(),
+                            move));
+            if (Objects.nonNull(add.defaultValue())) {
```

Comment:
`add.defaultValue()` is only working for Spark 3.4.

Reply:
`UpdateColumnDefaultValue` also exists in spark 3.4, how should I deal with this situation？
is to close this PR Or is it written under the spark3.4 module?

Reply:
It is difficult to support both 3.3 and 3.4.
I think we can just close this PR now. Until we really need this feature.

Reply:
Thanks.